spark/src/main/scala/geotrellis/spark/io/hadoop/Implicits.scala

```diff
@@ -16,35 +16,107 @@
 
 package geotrellis.spark.io.hadoop
 
+import geotrellis.tiling.SpatialComponent
 import geotrellis.raster.CellGrid
 import geotrellis.raster.io.geotiff.GeoTiff
-import geotrellis.raster.io.geotiff.writer.GeoTiffWriter
 import geotrellis.raster.render.{Jpg, Png}
+import geotrellis.raster.resample._
+import geotrellis.layers._
+import geotrellis.layers.avro._
+import geotrellis.layers.hadoop._
+import geotrellis.spark.io._
+import geotrellis.spark.util.KryoSerializer
+import geotrellis.util.MethodExtensions
+
+import spray.json._
+import spray.json.DefaultJsonProtocol._
 
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.Path
 import org.apache.spark._
 import org.apache.spark.rdd._
 
+import scala.reflect.ClassTag
+
+
 object Implicits extends Implicits
 
 trait Implicits {
   implicit class HadoopSparkContextMethodsWrapper(val sc: SparkContext) extends HadoopSparkContextMethods
   implicit class withSaveBytesToHadoopMethods[K](rdd: RDD[(K, Array[Byte])]) extends SaveBytesToHadoopMethods[K](rdd)
   implicit class withSaveToHadoopMethods[K,V](rdd: RDD[(K,V)]) extends SaveToHadoopMethods[K, V](rdd)
 
-  implicit class withJpgHadoopWriteMethods(val self: Jpg) extends HadoopRasterMethods[Jpg] {
-    def write(path: Path, conf: Configuration): Unit =
-      HdfsUtils.write(path, conf) { _.write(self.bytes) }
+  implicit class withJpgHadoopSparkWriteMethods(val self: Jpg) extends JpgHadoopSparkWriteMethods(self)
+
+  implicit class withPngHadoopSparkWriteMethods(val self: Png) extends PngHadoopSparkWriteMethods(self)
+
+  implicit class withGeoTiffSparkHadoopWriteMethods[T <: CellGrid[Int]](val self: GeoTiff[T]) extends GeoTiffHadoopSparkWriteMethods[T](self)
+
+  implicit class withHadoopAttributeStoreMethods(val self: HadoopAttributeStore.type) extends MethodExtensions[HadoopAttributeStore.type] {
```

Comment:
I know that we removed implicit methods like these before, but these work; and they allow for convenient `apply` methods without breaking preexisting code

Reply:
This is implicit extension to add `.write(path: Path)` for writing through hadoop FS, which I didn't know we had, but make sense. What are the `apply` methods in question?

Reply:
Oh, I was referring to `withHadoopAttributeStoreMethods` and similar classes that add `apply` methods that utilize the `SparkContext`.